src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,197 @@
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dbschema.Versioned;
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.PgUtil;
+import org.folio.rest.persist.PostgresClient;
+
+public class ShelvingOrderUpdate {
+
+  private static final Logger log = LogManager.getLogger();
+
+  private static final String SQL_EXISTS_ITEMS_TO_UPDATE = new StringBuilder()
+      .append("SELECT EXISTS (SELECT null FROM item WHERE not (jsonb ? 'effectiveShelvingOrder'))")
+      .toString();
+
+  private static final String SQL_SELECT_ITEMS = new StringBuilder()
+      .append("SELECT id AS \"id\", ")
+      .append("jsonb AS \"itemJson\" ")
+      .append("FROM item WHERE not (jsonb ? 'effectiveShelvingOrder') ")
+      .append(" FOR UPDATE ")
+      .append("LIMIT $1").toString();
+
+
+  private static final String SQL_UPDATE_ITEMS = new StringBuilder()
+      .append("UPDATE ITEM ")
+      .append("SET jsonb = $1 ")
+      .append("WHERE id = $2").toString();
+
+  // Defines version threshold where this property was introduces from
+  private static final String FROM_VERSION_SHELVING_ORDER = "20.1.0";
+
+  private static final String OPERATION_STATUS_VERSION_MISMATCH = "Module version isn't expected";
+  private static final String OPERATION_STATUS_ITEMS_UPDATES_SUCCESSFUL = "Items updates completed successfully";
+  private static final String OPERATION_STATUS_ITEMS_UPDATES_FAILED = "Items updates failed";
+
+  private Versioned versionChecker;
+
+  private int updateBunchSize;
+  private Map<UUID, String> itemsShelvingOrderMap;
+  private Map<UUID, Item> itemsMap;
+
+  public ShelvingOrderUpdate(int updateBunchSize) {
+    this.versionChecker = new Versioned() { };
+    this.versionChecker.setFromModuleVersion(FROM_VERSION_SHELVING_ORDER);
+    this.updateBunchSize = updateBunchSize;
+
+    itemsShelvingOrderMap = new LinkedHashMap<>();
+    itemsMap = new LinkedHashMap<>();
+  }
+
+  public Future<String> updateItems(TenantAttributes attributes, Map<String, String> headers, Context vertxContext) {
+    Promise<String> promise = Promise.promise();
+
+    String fromModuleVersion = attributes.getModuleFrom().trim();
+    if (StringUtils.isBlank(fromModuleVersion) || versionChecker.isNewForThisInstall(fromModuleVersion)) {
+      // Update items only for versions before the expected
+      promise.fail(OPERATION_STATUS_VERSION_MISMATCH);
+    } else {
+      PostgresClient postgresClient =  PgUtil.postgresClient(vertxContext, headers);
+
+      hasItemsToUpdate(vertxContext.owner(), postgresClient)
+          .onSuccess(itemsExists -> {
+            log.warn("Items update started");
+          }).onFailure(h -> {
+            log.info("No items for update");
+      });
```

Comment:
Error ignored here. I guess that was intented.

Reply:
Here just only messages logged, but it looks in case of failure we need to fail upgrade part of tenant loading. Not sure if it possibly to separate it from rest of loadData. Therefore there is only logging remain. How do you think, does it make sense?